sonic-net/sonic-buildimage
# SONiC bootloader signing for UEFI secureboot

SONiC bootloader signing for UEFI secureboot

Signed-off-by: Sachin Naik <sachnaik@cisco.com>



#### Why I did it
    This enables build framework to sign EFI bootloader images such as Grubx64.efi and Shimx64.efi.
    These signed bootloader gets verified on secure boot enabled system. The systems secure boot chain
    of trust ensures that it runs only verifiable boot loaders.

    The secure boot chain of trust verifies images as below
    UEFI FW -> SHIM -> GRUB -> Linux Kernel

    Currently the UEFI signing is supported at Linux kernel.
    This fix fills gap of signing and verification path between UEFI FW and Linux kernel.

#### How I did it
    1. Added Debian GRUB2 & SHIM source package to the SONiC build system.
    2. Added signing hook in build_debian.sh for signing GRUB2 and SHIM efi image.
    3. Defined build configurable hook in rules/config, for secure boot configuration.
    4. By default signing hook uses sbsign-tool for signing these EFI images.
    5. Added framework to configure build system to integrate SONiC vendor's specific signing tools.
    6. Added installer.sh to install signed SHIM and GRUB in EFI system partition.

#### How to verify it
Build SONiC image with UEFI secure boot enable which produces signed SHIM, GRUB and Linux Kernel images 
Load them on UEFI secure boot enabled system. The necessary verification keys should be enrolled in the system for secure boot image verification.

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog

Mechanism in SONiC build systems to produce signed EFI images.

#### Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
